codingstandards/JS/README.md

```diff
@@ -1,13 +1,95 @@
-# GTMSportswear Javascript/TypeScript Coding Standards
+# GTMSportswear JavaScript/TypeScript Coding Standards
 
-*GTMSportswear approach to Javascript*
+*GTMSportswear's approach to JavaScript and TypeScript*
 
 ## Table of Contents
 
+1. [Why TypeScript?](#why-typescript)
+1. [TypeScript Compilation](#typescript-compilation)
 1. [TypeScript Entry Points](https://github.com/GTMSportswear/docs/blob/master/codingstandards/JS/typescriptentrypoints.md)
-1. TypeScript Modules
-1. TypeScript Api Services
+1. [TypeScript Modules](#typescript-modules)
+1. [Retrieving Data](#retrieving-data)
+1. Documentation Comments
 1. TypeScript Utilities
   1. Solution Utilities
   1. GitHub Utilities
-1. QUnit Testing
\ No newline at end of file
+1. Asynchronous TypeScript
+1. Unit Testing
+1. Integration Testing
+
+## Why TypeScript?
+A few years ago, we made a team decision to move away from pure JavaScript and instead use TypeScript. The main driver behind this was to allow us to add stricter type checking to our code and give us better intellisense during the coding process.
```

Comment:
@samiam236 I wouldn't reference this as "A few years ago" if this sticks a couple years later then this reference doesn't work here. I would just state it was a team decision.

Reply:
Good call.